File: Java/1984. Minimum Difference Between Highest and Lowest of K Scores.java
class Solution {
    public int minimumDifference(int[] nums, int k) {
        Arrays.sort(nums);
        int minDiff = nums[nums.length - 1] - nums[0];
        for (int r = 0; r < nums.length; r++) {
            if (r - k + 1 >= 0) {
                minDiff = Math.min(minDiff, nums[r] - nums[r - k + 1]);
            }
        }
        return minDiff;
    }
}

File: Python3/1984. Minimum Difference Between Highest and Lowest of K Scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        l, min_diff = 0, nums[-1] - nums[0]

        for r, num in enumerate(nums):
            if r - l + 1 == k:
                min_diff = min(min_diff, num - nums[l])
                l += 1

        return min_diff